### ISSUE TITLE
k8s 1.12/k8s1.13  Ingress pointing to daemonset is not functional. "503 Service Temporarily Unavailable" error is seen when accessing ingress.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#16391]
Rancher server version - v2.1.2-rc2

Set up - DO cluster with 3 worker nodes and 1 etcd/control node.

Steps to reproduce the problem:
Create a  daemonset .
Create an ingress pointing to this daemonset .

Ingress pointing to daemonset is not functional. "503 Service Temporarily Unavailable" error is seen when accessing ingress.

Note - Kubernetes version was 1.12 in this case.

Ingress controller logs:
```
I1031 18:28:22.813927 6 controller.go:179] Backend successfully reloaded.
I1031 18:28:57.819868 6 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"test-56049", Name:"default-55860", UID:"d485d3ef-dd3a-11e8-a517-d6487b1d788b", APIVersion:"extensions", ResourceVersion:"1670", FieldPath:""}): type: 'Normal' reason: 'CREATE' Ingress test-56049/default-55860
W1031 18:28:58.765641 6 controller.go:769] Service "test-56049/ingress-967984637dbc9be8bea1e7e598ec0774" does not have any active Endpoint.
I1031 18:28:58.765947 6 controller.go:169] Configuration changes detected, backend reload required.
I1031 18:28:58.880246 6 controller.go:179] Backend successfully reloaded.
W1031 18:29:02.098931 6 controller.go:769] Service "test-56049/ingress-967984637dbc9be8bea1e7e598ec0774" does not have any active Endpoint.
W1031 18:29:05.641536 6 controller.go:769] Service "test-56049/ingress-967984637dbc9be8bea1e7e598ec0774" does not have any active Endpoint.
I1031 18:29:05.641830 6 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"test-56049", Name:"default-55860", UID:"d485d3ef-dd3a-11e8-a517-d6487b1d788b", APIVersion:"extensions", ResourceVersion:"1697", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress test-56049/default-55860
I1031 18:29:05.651541 6 event.go:218] Event(v1.ObjectReference{Kind:"Ingress", Namespace:"test-56049", Name:"default-55860", UID:"d485d3ef-dd3a-11e8-a517-d6487b1d788b", APIVersion:"extensions", ResourceVersion:"1701", FieldPath:""}): type: 'Normal' reason: 'UPDATE' Ingress test-56049/default-55860
W1031 18:29:08.975173 6 controller.go:769] Service "test-56049/ingress-967984637dbc9be8bea1e7e598ec0774" does not have any active Endpoint.
63.205.203.157 - [63.205.203.157] - - [31/Oct/2018:18:29:09 +0000] "GET /name.html HTTP/1.1" 503 214 "-" "curl/7.43.0" 86 0.000 [] - - - - 6c0ebf10dadc67b75f03842944f5d3d5
63.205.203.157 - [63.205.203.157] - - [31/Oct/2018:18:41:05 +0000] "GET /name.html HTTP/1.1" 503 214 "-" "curl/7.43.0" 86 0.000 [] - - - - 1ae991a544d8f27ef7b60c48b4008acf
```

**Comment by soumyalj** [comment]
https://github.com/rancher/rancher/issues/17309

**Comment by alena1108** [comment]
@bmdepesa @sangeethah can you confirm that this issue happens on master branch

**Comment by jiaqiluo** [comment]
The same bug exists for the cluster k8s v1.12.4, docker 18.9.1 on rancher 2.1.6

**Comment by sowmyav27** [comment]
Able to reproduce this in rancher:v2.1.7-rc2, k8s v1.12.5.

**Comment by soumyalj** [comment]
This issue can be reproduced in the below configuration:
rancher version:v2.1.7-rc2 with k8s v1.13.1

**Comment by alena1108** [comment]
Happens due to k8s 1.10 being a base version for rancher and norman (vendor.conf). Have to switch to 1.12 base

**Comment by alena1108** [comment]
Resolved in `rancher/rancher:v2.1.7-rc4`

**Comment by sowmyav27** [comment]
verified with `rancher/rancher:v2.1.7-rc4`. 

verified on k8s v1.13.1 and v1.12.6. Ingress pointing to daemonset are functional and are access all the pods in the workload.
